### PR TITLE
Updating to RxSwift 5.

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Just updating to the new version :))

### CODE REVIEW

**Comment by freak4pc** [comment]
Seems to be conflicting @LucianoPAlmeida

**Comment by LucianoPAlmeida** [comment]
Hey @freak4pc :)
What do you mean by conflicting?

**Comment by freak4pc** [comment]
@LucianoPAlmeida 
image

**Comment by freak4pc** [comment]
Can you rebase on top of master?

**Comment by LucianoPAlmeida** [comment]
That weird this is what I'm seeing haha
Screen Shot 2019-05-01 at 13 02 16

**Comment by freak4pc** [comment]
Becuase you're doing a merge and I'm trying a rebase :)

**Comment by LucianoPAlmeida** [comment]
Huum sorry, didn't notice, rebasing :)

File: Action.podspec [hunk 1]
```diff
@@ -20,8 +20,8 @@ Pod::Spec.new do |s|
   s.source_files  = "Sources/**/*.{swift}"
 
   s.frameworks  = "Foundation"
-  s.dependency "RxSwift", "~> 4.0"
-  s.dependency "RxCocoa", "~> 4.0"
+  s.dependency "RxSwift", "~> 5.0.0"
```
Comment (freak4pc) [inline Action.podspec]: Let's do `~> 5.0` everywhere

File: Tests/ActionTests/ActionTests.swift [hunk 2]
```diff
@@ -31,8 +31,8 @@ class ActionTests: QuickSpec {
                 action.inputs.bind(to: inputs).disposed(by: disposeBag)
                 scheduler.start()
                 XCTAssertEqual(inputs.events, [
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: FYI you can do

```
Recorded.events([
 .next(
 .next(
])
```

File: Tests/ActionTests/ActionTests.swift [hunk 3]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
-				
+
 				it("element receives single value for each execution") {
 					XCTAssertEqual(element.events, [
-						next(10, "a"),
-						completed(10),
-						next(20, "b"),
-						completed(20),
+						Recorded.next(10, "a"),
+                        Recorded.completed(10),
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift [hunk 4]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
-						next(10, "a"),
-						next(10, "a"),
-						next(10, "a"),
-						completed(10),
-						next(20, "b"),
-						next(20, "b"),
-						next(20, "b"),
-						completed(20),
+						Recorded.next(10, "a"),
+                        Recorded.next(10, "a"),
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift [hunk 5]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
+
 					scheduler.start()
 				}
-				
+
 				it("first element receives single value") {
 					XCTAssertEqual(element.events, [
-						next(30, "a"),
-						completed(30),
+                        Recorded.next(30, "a"),
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift [hunk 6]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
+                        Recorded.completed(30)
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift [hunk 7]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
 						])
 				}
-				
+
 				it("second element fails with notEnabled error") {
 					XCTAssertEqual(secondElement.events, [
-						error(20, ActionError.notEnabled)
+                        Recorded.error(20, ActionError.notEnabled)
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift [hunk 8]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
 					bindAndExecuteTwice(action: action)
 				}
-				
+
 				it("element fails with notEnabled") {
 					XCTAssertEqual(element.events, [
-						error(10, ActionError.notEnabled),
-						error(20, ActionError.notEnabled),
+                        Recorded.error(10, ActionError.notEnabled),
+                        Recorded.error(20, ActionError.notEnabled)
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift [hunk 9]
```diff
@@ -107,479 +107,479 @@ class ActionTests: QuickSpec {
...
 					action = Action { _ in Observable.error(TestError) }
 					bindAndExecuteTwice(action: action)
 				}
-				
+
 				it("element fails with underlyingError") {
 					XCTAssertEqual(element.events, [
-						error(10, ActionError.underlyingError(TestError)),
-						error(20, ActionError.underlyingError(TestError)),
+                        Recorded.error(10, ActionError.underlyingError(TestError)),
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: misaligned

File: Tests/ActionTests/ActionTests.swift (hunk 3 above)
Comment (LucianoPAlmeida) [reply Tests/ActionTests/ActionTests.swift]: Thought swiftlint autocorrect would fix it :(

File: Tests/ActionTests/ActionTests.swift [hunk 10]
```diff
@@ -354,37 +367,41 @@ class ActionTests: QuickSpec {
 			describe("disabled action") {
 				sharedExamples("send notEnabled errors to errors observable") {
 					it("inputs subject receives generated inputs") {
-						XCTAssertEqual(inputs.events, [
-							Recorded.next(10, "a"),
-							Recorded.next(20, "b")
-							])
+						XCTAssertEqual(inputs.events, Recorded.events([
+                            .next(10, "a"),
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: It's still misaligned IMO... Can you make sure all of the events are indented after the `XCT` portion?

Comment (LucianoPAlmeida) [reply Tests/ActionTests/ActionTests.swift]: yeah, for sure. I thought crtl + i on Xcode would solve it all, but apparently not haha

Comment (LucianoPAlmeida) [reply Tests/ActionTests/ActionTests.swift]: Now, I notice that is the default spacing of xcode, should we change it then? or just align this part?

Comment (freak4pc) [reply Tests/ActionTests/ActionTests.swift]: Default Xcode alignment is crap :P Let's do the one that makes sense (after XCT)_

Comment (LucianoPAlmeida) [reply Tests/ActionTests/ActionTests.swift]: haha Sure :) Done

**Comment by freak4pc** [comment]
I need to resolve this on master first: https://github.com/RxSwiftCommunity/Action/pull/193

**Comment by LucianoPAlmeida** [comment]
Sure @freak4pc, but anything left on this one?

**Comment by freak4pc** [comment]
There will be issues here once I merge this because we removed InputSubject, for example. You'll have to rebase again after, hopefully It'd be fine :)

**Comment by LucianoPAlmeida** [comment]
Nice @freak4pc, thank's for the review :))

**Comment by freak4pc** [comment]
@LucianoPAlmeida I think that would do it. Can you rebase one final time on master and resolve any conflicts ? Thanks.

**Comment by LucianoPAlmeida** [comment]
@freak4pc  Rebase done \o/

**Comment by freak4pc** [comment]
@LucianoPAlmeida Mind paying back with a review? ^_^ 
https://github.com/RxSwiftCommunity/RxSwiftExt/pull/205

**Comment by LucianoPAlmeida** [comment]
Sure :))

**Comment by freak4pc** [comment]
Already got a review there haha. Thank you!

**Comment by LucianoPAlmeida** [comment]
Haha no problem, feel free to ping me anytime. I’ll be happy if I can help :)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**